sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -191,6 +195,21 @@ def on_challenge(
         :rtype: bool
         """
         # pylint:disable=unused-argument
+        headers = response.http_response.headers
+        error = get_challenge_parameter(headers, "Bearer", "error")
+        if error == "insufficient_claims":
+            encoded_claims = get_challenge_parameter(headers, "Bearer", "claims")
+            if not encoded_claims:
+                return False
+            claims = base64.urlsafe_b64decode(encoded_claims).decode("utf-8")
+            if claims:
+                try:
+                    token = self._get_token(*self._scopes, claims=claims)
+                    bearer_token = cast(Union["AccessToken", "AccessTokenInfo"], token).token
+                    self._update_headers(request.http_request.headers, bearer_token)
```

Comment:
I see we are not caching the token retrieved in this case (i.e. using `self.authorize_request(request, *self._scopes, claims=claims)`). Is this intentional, and we expect another get_token call on subsequent requests?